scripts/Inelastic/IndirectReductionCommon.py

```diff
@@ -592,12 +592,11 @@ def get_group_from_string(grouping_string):
         return [int(grouping_string)]
 
 
-def create_group_from_string(input_workspace, grouping_string):
-    from mantid.simpleapi import GroupDetectors
-    return GroupDetectors(InputWorkspace=input_workspace,
-                          Behaviour='Average',
-                          SpectraList=get_group_from_string(grouping_string),
-                          StoreInADS=False)
+def create_group_from_string(group_detectors, grouping_string):
+    group_detectors.setProperty("WorkspaceIndexList", get_group_from_string(grouping_string))
+    group_detectors.setProperty("OutputWorkspace", "__temp")
+    group_detectors.execute()
+    return group_detectors.getProperty("OutputWorkspace").value
```

Comment:
I see you return it but please correct me if I am wrong (I don't use python too frequently) This stores something in the ADS that hidden or not, is not removed?

Reply:
@Pasarus `group_detectors` is used as a child algorithm, and so it does not store any outputs in the ADS

Reply:
:+1: